prov/efa/src/rdm/efa_rdm_pke.c

```diff
@@ -392,70 +386,65 @@ ssize_t efa_rdm_pke_sendv(struct efa_rdm_ep *ep,
 	conn = efa_av_addr_to_conn(ep->base_ep.av, pkt_entry_vec[0]->addr);
 	assert(conn && conn->ep_addr);
 
-	prev_send_wr = NULL;
-	for (i = 0; i < pkt_entry_cnt; ++i) {
-		pkt_entry = pkt_entry_vec[i];
+	qp = ep->base_ep.qp;
+	ibv_wr_start(qp->ibv_qp_ex);
+	for (pkt_idx = 0; pkt_idx < pkt_entry_cnt; ++pkt_idx) {
+		pkt_entry = pkt_entry_vec[pkt_idx];
```

Comment:
Do we need (or already hold?) a lock to use ibv here?

Reply:
No. rdma-core is thread safe. The manual of `ibv_wr_start` specifically called out that it will obtain lock.